building-applications/vector-databases.md

```diff
@@ -1,26 +1,30 @@
 # Vector Databases
 
-Vector databases are a crucial part of the generative AI workflow. They’re particularly suited for similarity searches, where an application searches for similar items like text, images, or audio.&#x20;
+Vector databases are a crucial part of the generative AI workflow. Vector databases store vector representations (embeddings) of text, images, sound, etc. They include search tools to enable similarity search across the vector representations to find semanitically similar data.   
 
-Vectorization feeds text to a machine learning algorithm, which vectorizes the data so that the more similar the words are, the closer their vectors. This creates context for the search query, instead of just trying to match a string, and returns more useful results.&#x20;
+Vector databases are typically used as part of retrieval augmented generation (RAG). In the RAG workflow, relevant documents or passages are retrieved from a vector database based on their semantic relevance. These documents or passages are then included in the LLM prompt to provide additional context for the LLM to use when generating a response. This pattern is used to:
 
-LangStream currently has native support for [DataStax Astra DB](https://www.datastax.com/products/datastax-astra), [Pinecone](https://www.pinecone.io/)
+* Reduce AI hallucinations
+* Provide more accurate, up-to-date, and context-aware responses
+* Extend the knowledge base of the LLM
+
+LangStream makes it easy to build applications using the RAG pattern. It currently has native support for [DataStax Astra DB](https://www.datastax.com/products/vector-search), [Pinecone](https://www.pinecone.io/), [Milvus/Zilliz](https://milvus.io/)
 and [Apache Cassandra](https://cassandra.apache.org).&#x20;
 
-Check out the [vector-db-sink agent](../pipeline-agents/input-and-output/vector-db-sink.md) and the [query-vector-db agent](../pipeline-agents/text-processors/query-vector-db.md) for more information on how to use vector databases in your LangStream application.&#x20;
+When working with a vector database you will either be writing vector embeddings to a vector database or performing semantic similarity queries across the vectors in the database. Check out the [vector-db-sink agent](../pipeline-agents/input-and-output/vector-db-sink.md) for writing to vector databases and the [query-vector-db agent](../pipeline-agents/text-processors/query-vector-db.md) for querying.&#x20;
 
-There is also built-in support for querying any Database which has a [JDBC](https://docs.oracle.com/javase/tutorial/jdbc/overview/index.html) driver.
+There is also built-in support for querying any database which has a [JDBC](https://docs.oracle.com/javase/tutorial/jdbc/overview/index.html) driver.
```

Comment:
Now you can also write to JDBC databases, maybe we can take this chance to update this paragraph

Reply:
Added